engines/agos/event.cpp

```diff
@@ -503,6 +503,35 @@ void AGOSEngine::delay(uint amount) {
 						GUI::AboutDialog aboutDialog;
 						aboutDialog.runModal();
 					}
+				} else if (event.kbd.keycode == Common::KEYCODE_f && getGameId() == GID_WAXWORKS) {
+					// Check if the necessary HitAreas exist.
+					HitArea *fightButton = findBox(117);
+					HitArea *walkButton = findBox(119);
```

Comment:
You don't need both hit areas here. You can toggle normal and fighting mode with the fight (the sword) button.
So, the rest of the code can be greatly simplified.